vignettes/ae-exp-adj.Rmd

```diff
@@ -0,0 +1,347 @@
+library(dplyr)
+```
+ 
+
+####  1. Data pretreatment 
+
+```{r,echo=FALSE,warning=FALSE, message=FALSE}
+# ADSL
+adsl <- r2rtf::r2rtf_adsl
+adsl$TRTA <- adsl$TRT01A
+adsl$TRTA <- factor(adsl$TRTA,
+                    levels = c("Placebo", "Xanomeline Low Dose", "Xanomeline High Dose"),
+                    labels = c("Placebo", "Low Dose", "High Dose"))
+# ADAE
+adae <- r2rtf::r2rtf_adae
+adae$TRTA <- factor(adae$TRTA,
+                    levels = c("Placebo", "Xanomeline Low Dose", "Xanomeline High Dose"),
+                    labels = c("Placebo", "Low Dose", "High Dose"))
+
+# Drug-related AE values
+adae$related <- ifelse(
+  adae$AEREL == "RELATED",
+  "Y",
+  ifelse(
+    toupper(adae$AEREL) == "NOT RELATED",
+    "N",
+    tools::toTitleCase(tolower(adae$AEREL))
+    )
+)
+
+# AE outcome
+for (i in seq_along(adae$AEOUT)) {
+  adae$outcome <- switch(adae$AEOUT[i],
+                         "RECOVERED/RESOLVED" = "Resolved",
+                         "RECOVERING/RESOLVING" = "Resolving",
+                         "RECOVERED/RESOLVED WITH SEQUELAE" = "Sequelae",
+                         "NOT RECOVERED/NOT RESOLVED" = "Not Resolved",
+                         tools::toTitleCase(tolower(adae$AEOUT[i]))
+                         )
+}
+
+# AE action
+adae$AEACN <- gsub("", "DOSE NOT CHANGED", adae$AEACN)
+
+for (i in seq_along(adae$AEACN)) {
+  adae$action_taken[i] <- switch(adae$AEACN[i],
+                                 "DOSE NOT CHANGED" = "None",
+                                 "DOSE REDUCED" = "Reduced",
+                                 "DRUG INTERRUPTED" = "Interrupted",
+                                 "DOSE INCREASED" = "Increased",
+                                 "NOT APPLICABLE" = "N/A",
+                                 "UNKNOWN" = "Unknown",
+                                 "''" = "None",
+                                 tools::toTitleCase(tolower(adae$AEACN[i]))
+                                 )
+}
+
+# AE duration with unit
+adae$duration <- paste(
+  ifelse(
+    is.na(adae$ADURN),
+    "",
+    as.character(adae$ADURN)
+    ),
+  tools::toTitleCase(tolower(adae$ADURU)),
+  sep = " "
+)
+
+for (i in seq_along(adae$duration)) {
+  if (is.na(adae$ADURN[i])) {
+    adae$duration[i] <- ifelse(
+      charmatch(toupper(adae$AEOUT[i]), "RECOVERING/RESOLVING") > 0 |
+        charmatch(toupper(adae$AEOUT[i]), "NOT RECOVERED/NOT RESOLVED") > 0,
+      "Continuing",
+      "Unknown"
+      )
+  }
+}
+
+# Assign label
+adae <- metalite::assign_label(adae,
+                               var = c("related", "outcome", "duration", "AESEV", "AESER", "AEDECOD", "action_taken"),
+                               label = c("Related", "Outcome", "Duration", "Intensity", "Serious", "Adverse Event", "Action Taken")
+)
+```
+At the end of this step, adae is created.(See .Rmd file for codes)
+```{r,include=FALSE}
+head(adae)
+```
+
+####  2. Build a metadata 
+
+```{r,echo=FALSE,warning=FALSE, message=FALSE}
+plan <- plan(

```

Comment:
Instead of creating a new metalite object, could we update `meta_ae_example.R` ? 

https://github.com/Merck/metalite.ae/blob/main/R/meta_ae_example.R

Reply:
Agree. I built this new meta example in Rmd file for Bing's development. But we should merge this example to `meta_ae_example.r` in the future.